src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -718,10 +720,64 @@ InterpreterSelectQuery::InterpreterSelectQuery(
     sanitizeBlock(result_header, true);
 }
 
+void InterpreterSelectQuery::executePutInCache(QueryPlan & query_plan)
+{
+    auto settings = context->getSettingsRef();
+    QueryCachePtr query_cache = context->getQueryCache();
+    if (!settings.query_cache_active_usage || query_cache == nullptr)
```

Comment:
I guess if the cache is currently `nullptr` (because the database started with `query_cache_active_usage = false`)  and the user just configured `query_cache_active_usage = true`, then we should create the cache and not silently return.